packages/cozy-client/src/StackLink.js

```diff
@@ -1,8 +1,26 @@
 import { MutationTypes } from './queries/dsl'
 import CozyLink from './CozyLink'
+import { CozyClientDocument } from './types'
 
 /**
+ * Returns full documents after a bulk update
  *
+ * @param  {{ ok: boolean, id: string, rev: string }[]} updateAllResponse - Response from bulk docs
+ * @param  {CozyClientDocument[]} originalDocuments - Documents that were updated
+ * @returns {{ data: CozyClientDocument[] }} - Full documents with updated _id and _rev
+ */
+const transformBulkDocsResponse = (updateAllResponse, originalDocuments) => {
+  return {
+    data: originalDocuments.map((od, i) => ({
+      ...od,
+      _id: updateAllResponse[i].id,
+      _rev: updateAllResponse[i].rev
+    }))
+  }
+}
```

Comment:
Are we sure that `CouchDB::_bulk_docs` returns the result in the same order as the as the payload we sent?

Reply:
Good question, this is not specified in the doc, but it would be a problem if it weren't since otherwise you would not be able to link newly created documents (without id) to their server part. The code seems to indicate that only maps are used so I think it should be OK: https://github.com/apache/couchdb/blob/0059b8f90e58e10b199a4b768a06a762d12a30d3/src/couch/src/couch_httpd_db.erl, I'll recheck tomorrow.

Reply:
Yep the list is in order https://github.com/apache/couchdb/blob/bf355b96ee3efc24cb6edfb3eeb5b6fe519fdbf8/src/chttpd/src/chttpd_db.erl#L514.